CONTRIBUTING.md

```diff
@@ -0,0 +1,71 @@
+# Contributing to Bolt
+
+Thank you for your interest in contributing to Bolt! 
+
+We welcome contributions at all levels. Whether you’re a beginner or an experienced developer, your input is valued. This guide will help you get started.
+
+If you have questions or need help, feel free to reach out to the team on [Discord](https://discord.com/invite/Q8xAsuCVrT).
+
+---
+
+## Code of Conduct
+
+Bolt adheres to the [Rust Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct). Please ensure your interactions are respectful and constructive.  
+
+Violations of the Code of Conduct can be reported by contacting the team.
+
+---
+
+## Ways to Contribute
+
+There are several ways you can contribute:
+
+1. **Report an Issue**  
+   If you’ve found a bug or have feedback, open an issue in the issue tracker. Be sure to include relevant details like your environment and steps to reproduce the issue.
+
+2. **Contribute Code**  
+   Fix bugs, improve performance, or add new features by submitting a pull request.
+
+3. **Resolving Issues**  
+   If you're interested in resolving an issue, please comment on the issue to let us know you're working on it. A good place to start looking for work is issues labeled with [good first issue](https://github.com/chainbound/bolt/issues?q=is%3Aissue%20state%3Aopen%20label%3AD-good-first-issue).
```

Comment:
When there is an issue, it would be good to first ask for being assigned to it while providing a plan on how to tackle the PR if necessary.